### PR TITLE
Replaced nlunit-test with pw_unit_test in src/lib/dnssd/tests/

### PR DESCRIPTION
**Pull request by mbknust**

### CODE REVIEW

File: src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp [hunk 1]
```diff
@@ -40,299 +39,262 @@ static SerializedQNameIterator AsSerializedQName(const uint8_t (&v)[N])
...
         static const uint8_t kOneItem[] = "\04test\00";
         SerializedQNameIterator it      = AsSerializedQName(kOneItem);
 
-        NL_TEST_ASSERT(inSuite, it.Next());
-        NL_TEST_ASSERT(inSuite, strcmp(it.Value(), "test") == 0);
-        NL_TEST_ASSERT(inSuite, !it.Next());
-        NL_TEST_ASSERT(inSuite, it.IsValid());
+        EXPECT_TRUE(it.Next());
+        EXPECT_EQ(strcmp(it.Value(), "test"), 0);
+        EXPECT_TRUE(!it.Next());
```
Comment (jlatusek) [inline src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp:50]: Replace `EXPECT_TRUE(!...` with `EXPECT_FALSE`

File: src/lib/dnssd/minimal_mdns/tests/TestRecordData.cpp [hunk 2]
```diff
@@ -234,32 +232,11 @@ void TxtRecord(nlTestSuite * inSuite, void * inContext)
...
 
-    NL_TEST_ASSERT(inSuite, ParseTxtRecord(BytesRange(record, record + sizeof(record)), &accumulator));
-    NL_TEST_ASSERT(inSuite, accumulator.Data().size() == 4);
-    NL_TEST_ASSERT(inSuite, (accumulator.Data()[0] == make_pair<std::string, std::string>("some", "")));
-    NL_TEST_ASSERT(inSuite, (accumulator.Data()[1] == make_pair<std::string, std::string>("foo", "bar")));
-    NL_TEST_ASSERT(inSuite, (accumulator.Data()[2] == make_pair<std::string, std::string>("x", "y=z")));
-    NL_TEST_ASSERT(inSuite, (accumulator.Data()[3] == make_pair<std::string, std::string>("a", "")));
+    EXPECT_TRUE(ParseTxtRecord(BytesRange(record, record + sizeof(record)), &accumulator));
+    EXPECT_EQ(accumulator.Data().size(), 4u);
+    EXPECT_TRUE((accumulator.Data()[0] == make_pair<std::string, std::string>("some", "")));
```
Comment (jlatusek) [inline src/lib/dnssd/minimal_mdns/tests/TestRecordData.cpp:237]: Replace with EXPECT_EQ

**Review by jlatusek — CHANGES_REQUESTED** [review]

**Review by andy31415 — COMMENTED** [review]
Can we reduce the scope to individual locations rather than entire dnssd, to have smaller incremental steps?

File: src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp [hunk 3]
```diff
@@ -40,299 +39,262 @@ static SerializedQNameIterator AsSerializedQName(const uint8_t (&v)[N])
...
     {
         static const uint8_t kOneItem[] = "\04test\00";
         SerializedQNameIterator it      = AsSerializedQName(kOneItem);
 
-        NL_TEST_ASSERT(inSuite, it.Next());
-        NL_TEST_ASSERT(inSuite, strcmp(it.Value(), "test") == 0);
-        NL_TEST_ASSERT(inSuite, !it.Next());
-        NL_TEST_ASSERT(inSuite, it.IsValid());
+        EXPECT_TRUE(it.Next());
+        EXPECT_EQ(strcmp(it.Value(), "test"), 0);
```
Comment (andy31415) [inline src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp:49]: use https://pigweed.dev/pw_unit_test/#c.EXPECT_STREQ

File: src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp (hunk 1 above)
Comment (mbknust) [reply src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp]: Done, now part of https://github.com/project-chip/connectedhomeip/pull/33069

File: src/lib/dnssd/minimal_mdns/tests/TestRecordData.cpp (hunk 2 above)
Comment (mbknust) [reply src/lib/dnssd/minimal_mdns/tests/TestRecordData.cpp]: Done, now part of #33068

File: src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp (hunk 3 above)
Comment (mbknust) [reply src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp]: Ok, I've replaced EXPECT_EQ with EXPECT_STREQ where it made sense.
This file in particular is now part of #33069

**Comment by mbknust** [comment]
> Can we reduce the scope to individual locations rather than entire dnssd, to have smaller incremental steps?

Ok, I've split it into a bunch of smaller PRs (they are referenced above).

The scope of this one has been reduced to cover only src/lib/dnssd/tests/

File: src/lib/dnssd/tests/TestActiveResolveAttempts.cpp [hunk 4]
```diff
@@ -43,50 +41,50 @@ Optional<ActiveResolveAttempts::ScheduledAttempt> ScheduledBrowse(const Dnssd::D
...
+    EXPECT_FALSE(attempts.NextScheduled().HasValue());
 
     // Adding a single peer should result in it being scheduled
 
     attempts.MarkPending(MakePeerId(1));
 
-    NL_TEST_ASSERT(inSuite, attempts.GetTimeUntilNextExpectedResponse() == Optional<Timeout>(0_ms32));
-    NL_TEST_ASSERT(inSuite, attempts.NextScheduled() == ScheduledPeer(1, true));
-    NL_TEST_ASSERT(inSuite, !attempts.NextScheduled().HasValue());
+    EXPECT_EQ(attempts.GetTimeUntilNextExpectedResponse(), Optional<Timeout>(0_ms32));
```
Comment (arkq) [inline src/lib/dnssd/tests/TestActiveResolveAttempts.cpp:59]: Maybe it's worth moving to `std::optional` here too (as a part of `chip::Optional` -> `std::optional` transition)?

**Review by arkq — APPROVED** [review]

File: src/lib/dnssd/tests/TestActiveResolveAttempts.cpp (hunk 4 above)
Comment (mbknust) [reply src/lib/dnssd/tests/TestActiveResolveAttempts.cpp:59]: I'd rather change this in a separate PR, to keep this one small.

**Review by jlatusek — APPROVED** [review]
LGTM

**Review by andy31415 — APPROVED** [review]

**Pull request merged**